src/test/java/bio/terra/service/filedata/DrsServiceTest.java

```diff
@@ -651,14 +667,91 @@ public void postAccessUrlForObjectIdInvalidPassport() {
         UnauthorizedException.class,
         () ->
             drsService.postAccessUrlForObjectId(
-                googleDrsObjectId, "gcp-passport-us-central1", drsPassportRequestModel));
+                googleDrsObjectId, "gcp-passport-us-central1", drsPassportRequestModel, null));
+  }
+
+  @Test
+  public void testSignGcpUrlNoProject() {
+    testSignGcpUrl(null, false);
+  }
+
+  @Test
+  public void testSignGcpUrlSpecifiedProject() {
+    testSignGcpUrl("myProject", false);
+  }
+
+  @Test
+  public void testSignGcpUrlSpecifiedProjectSelfHosted() {
+    testSignGcpUrl("myProject", true);
+  }
+
+  public void testSignGcpUrl(String billingProject, boolean selfHostedDataset) {
+    SnapshotSummaryModel snapshotSummary =
+        new SnapshotSummaryModel().id(snapshotId).selfHosted(selfHostedDataset);
+    Snapshot snapshot =
+        mockSnapshot(snapshotId, billingProfile.getId(), CloudPlatform.GCP, SNAPSHOT_DATA_PROJECT);
+    snapshot.getSourceDataset().getDatasetSummary().selfHosted(selfHostedDataset);
+
+    if (selfHostedDataset) {
+      String sourcePath = "gs://path/to/file.txt";
+      String datasetProject = snapshot.getSourceDataset().getProjectResource().getGoogleProjectId();
+      snapshotSummary.selfHosted(selfHostedDataset);
+      Storage storage = mock(Storage.class);
+      Bucket bucket = mock(Bucket.class);
+      when(bucket.getLocation()).thenReturn("us-central1");
+      when(storage.get(eq(BlobId.fromGsUtilUri(sourcePath).getBucket()), any())).thenReturn(bucket);
+      // This mock is needed in order to get the region from the bucket
+      when(gcsProjectFactory.getStorage(datasetProject)).thenReturn(storage);
+
+      when(samService.signUrlForBlob(
+              eq(TEST_USER), eq(billingProject), eq(sourcePath), eq(URL_TTL)))
```

Comment:
Since you are checking equality for all arguments, you don't need the `eq` argument matchers:

```
when(samService.signUrlForBlob(TEST_USER, billingProject, sourcePath, URL_TTL)
```